pkg/reaper/reaper.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reaper
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/googleinterns/cloudai-gcp-test-resource-reaper/pkg/clients"
+	"github.com/googleinterns/cloudai-gcp-test-resource-reaper/pkg/resources"
+	"github.com/googleinterns/cloudai-gcp-test-resource-reaper/reaperconfig"
+	"google.golang.org/api/option"
+)
+
+// Reaper represents the resource reaper for a single GCP project. The reaper will
+// run on a given schedule defined in cron time format.
+type Reaper struct {
+	UUID      string
+	ProjectID string
+	Watchlist []resources.WatchedResource
```

Comment:
Let's make this an array of pointers

Reply:
Fixed. Lots of files needed to be changed so that pointers are used consistently throughout project.